### PR TITLE
Update production profile for monitoring component

### PR DESCRIPTION
**Pull request by iamps5**
**Description**

Changes proposed in this pull request:

Updating the production profile for the monitoring component with the latest settings used on the SAP based managed offering

**Related issue(s)**

### CODE REVIEW

**Comment by iamps5** [comment]
@ebensom can you please approve this?

File: resources/monitoring/profile-evaluation.yaml [hunk 1]
```diff
@@ -3,13 +3,61 @@ prometheus:
...
+        memory: 4Gi
       requests:
         cpu: 75m
         memory: 1000Mi
     retention: 6h
     retentionSize: 1GB
 
+prometheusOperator:
+  namespaces:
+    releaseNamespace: "true"
```
Comment (ebensom) [inline resources/monitoring/profile-evaluation.yaml:15]: I think this is specific to runtimes (SKRs) only where we don't want other namespaces to be watched by prometheus-operator.

File: resources/monitoring/profile-evaluation.yaml [hunk 2]
```diff
@@ -3,13 +3,61 @@ prometheus:
...
+      protocol: TCP
+      targetPort: 15020
+  sidecar:
+    dashboards:
+      searchNamespace: "kyma-system"
+    datasources:
+      searchNamespace: "kyma-system"
+  kyma:
+    authProxy:
+      configDocsLink: "https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/3e4299cfd0884c428e6b4774225638e8.html"
```
[the remark this answers is not in the document]
Comment (a-thaler) [reply resources/monitoring/profile-evaluation.yaml:63]: Yes, the overrides are specific for SKRs and need to stay. The component will not be migrated to a module but instead will be deleted at some point

File: resources/monitoring/profile-evaluation.yaml [hunk 3]
```diff
@@ -3,13 +3,61 @@ prometheus:
...
+      requests:
+        cpu: 200m
+        memory: 768Mi
+      limits:
+        cpu: 1000m
+        memory: 8Gi
+
+defaultRules:
+  disabled:
+    PrometheusNotConnectedToAlertmanagers: "true"
```
[the remark this answers is not in the document]
Comment (iamps5) [reply resources/monitoring/profile-evaluation.yaml:32]: @ebensom then should I add `alertmanager: { enable: "false"}`, or remove the above rule

File: resources/monitoring/profile-evaluation.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (iamps5) [reply resources/monitoring/profile-evaluation.yaml:63]: okay, I'll remove all these overrides

**Review by a-thaler — CHANGES_REQUESTED** [review]
That are mainly incompatible changes (like desabling the alertmanager by default), so the behaviour of the component will change fundamentally. If we think it is time to do such changes even before the removal of the component, we will do that in a controlled way. Please do not just apply changes to the open source project out of nothing.
As I said in a different comment, we do not plan to transform the component into a module.

**Comment by iamps5** [comment]
> That are mainly incompatible changes (like desabling the alertmanager by default), so the behaviour of the component will change fundamentally. If we think it is time to do such changes even before the removal of the component, we will do that in a controlled way. Please do not just apply changes to the open source project out of nothing. As I said in a different comment, we do not plan to transform the component into a module.

Okay, I'll remove the `deafultRules` section

**Comment by iamps5** [comment]
@a-thaler can you please review this now?

File: resources/monitoring/profile-production.yaml [hunk 4]
```diff
@@ -35,4 +51,30 @@ kube-state-metrics:
...
+      protocol: TCP
+      targetPort: 15020
+  sidecar:
+    dashboards:
+      searchNamespace: "kyma-system"
+    datasources:
+      searchNamespace: "kyma-system"
+  kyma:
+    authProxy:
+      configDocsLink: "https://help.sap.com/docs/BTP/65de2977205c403bbc107264b8eccf4b/3e4299cfd0884c428e6b4774225638e8.html"
```
Comment (a-thaler) [inline resources/monitoring/profile-production.yaml:80]: That whole block should not get in as it is specific to the SAP setup and will break compatibility from OS perspective

File: resources/monitoring/profile-production.yaml [hunk 5]
```diff
@@ -17,13 +17,29 @@ prometheus:
               storage: 20Gi
 
 prometheusOperator:
+  namespaces:
+    releaseNamespace: "true"
```
Comment (a-thaler) [inline resources/monitoring/profile-production.yaml:21]: That block should not get in as it is specific to the SAP setup and will break compatibility from OS perspective

File: resources/monitoring/profile-production.yaml [hunk 6]
```diff
@@ -17,13 +17,29 @@ prometheus:
...
+      requests:
+        cpu: 200m
+        memory: 768Mi
+      limits:
+        cpu: 1000m
+        memory: 8Gi
+
+defaultRules:
+  disabled:
+    PrometheusNotConnectedToAlertmanagers: "true"
```
Comment (a-thaler) [inline resources/monitoring/profile-production.yaml:41]: That whole block should not get in as it is specific to the SAP setup and will break compatibility from OS perspective

**Comment by iamps5** [comment]
@a-thaler My bad yesterday I made changes only in evaluation. Now, I have added recommend changes in production profile also. Please review

File: resources/logging/profile-evaluation.yaml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 loki:
   resources:
     limits:
-      cpu: 100m
```
Comment (a-thaler) [inline resources/logging/profile-evaluation.yaml:4]: The logging component got already removed from main branch

File: resources/monitoring/profile-evaluation.yaml [hunk 8]
```diff
@@ -3,13 +3,19 @@ prometheus:
...
         cpu: 600m
-        memory: 2000Mi
+        memory: 4Gi
       requests:
         cpu: 75m
         memory: 1000Mi
     retention: 6h
     retentionSize: 1GB
 
-grafana:
```
[the remark this answers is not in the document]
Comment (a-thaler) [reply resources/monitoring/profile-evaluation.yaml:13]: I agree, why we should remove it?

[the remark this answers is not in the document]
Comment (iamps5) [reply resources/monitoring/profile-evaluation.yaml:13]: Added back

File: resources/logging/profile-evaluation.yaml (hunk 7 above)
Comment (iamps5) [reply resources/logging/profile-evaluation.yaml:4]: removed

**Review by a-thaler — APPROVED** [review]

**Pull request merged**